src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -181,13 +181,14 @@ impl<'a> JobQueue<'a> {
                     match result {
                         Ok(()) => self.finish(key, cx)?,
                         Err(e) => {
+                            cx.config.shell().error(format!("{}", e))?;
```

Comment:
Could this use Cargo's top level error printing functionality? That'll be sure to print the whole stack trace of the error. Basically something along the lines of [calling these functions](https://github.com/rust-lang/cargo/blob/cf4986dd4690591aead3edc9b42275430f7640c4/src/cargo/lib.rs#L153-L202) (but perhaps not exactly).